classes/UpgradeSelfCheck.php

```diff
@@ -393,43 +406,60 @@ public function getRequirementWording(int $requirement, bool $isWebVersion = fal
     /**
      * @return string[]
      */
-    public function getCoreMissingFiles(): array
+    public function getCoreMissingFiles(): ?array
     {
         $missingFiles = $this->checksumCompare->getTamperedFilesOnShop($this->currentVersion);
+        if (!$missingFiles) {
+            return null;
+        }
 
         return array_merge($missingFiles['core']['missing'], $missingFiles['mail']['missing']);
     }
 
     /**
      * @return string[]
      */
-    public function getCoreAlteredFiles(): array
+    public function getCoreAlteredFiles(): ?array
     {
         $alteredFiles = $this->checksumCompare->getTamperedFilesOnShop($this->currentVersion);
+        if (!$alteredFiles) {
+            return null;
+        }
 
         return array_merge($alteredFiles['core']['altered'], $alteredFiles['mail']['altered']);
     }
 
     /**
      * @return string[]
      */
-    public function getThemeMissingFiles(): array
+    public function getThemeMissingFiles(): ?array
     {
         $missingFiles = $this->checksumCompare->getTamperedFilesOnShop($this->currentVersion);
+        if (!$missingFiles) {
+            return null;
+        }
 
         return $missingFiles['themes']['missing'];
     }
 
     /**
      * @return string[]
      */
-    public function getThemeAlteredFiles(): array
+    public function getThemeAlteredFiles(): ?array
     {
         $alteredFiles = $this->checksumCompare->getTamperedFilesOnShop($this->currentVersion);
+        if (!$alteredFiles) {
+            return null;
+        }
 
         return $alteredFiles['themes']['altered'];
     }
 
+    public function isAlteredFilesNull(): bool
+    {
+        return $this->getCoreAlteredFiles() === null || $this->getCoreMissingFiles() || $this->getThemeAlteredFiles() === null || $this->getThemeMissingFiles();
```

Comment:
```suggestion
        return $this->getCoreAlteredFiles() === null
            || $this->getCoreMissingFiles() === null
            || $this->getThemeAlteredFiles() === null
            || $this->getThemeMissingFiles() === null;
```

Some of the conditions seemed to lack their `=== null`, I let you check if this was expected.